test/uq_test_parameters.toml

```diff
@@ -0,0 +1,82 @@
+# The six parameters below are to be used for calibration / uncertainty 
+# quantification
+[uq_param_1]
+prior = "Parameterized(Normal(2.0, 1.0))"
+constraint = "no_constraint()"
+type = "float"
+
+[uq_param_2]
+prior = "Parameterized(Gamma(5.0, 2.0))"
+constraint = "bounded_below(3.0)"
+type = "float"
+
+[uq_param_3]
+prior = "Parameterized(MvNormal(4, 0.1))"
+constraint = "[no_constraint(), bounded_below(-1.0), bounded_above(0.4), bounded(-0.1,0.2)]"
+type = "float"
+
+[uq_param_4]
+prior = "Samples([1.0 3.0 5.0 7.0; 9.0 11.0 13.0 15.0])"
+constraint = "[bounded(10.0, 15.0), bounded_below(-1.0)]"
+type = "float"
+
+[uq_param_5]
+prior = "Samples([1.0 3.0; 5.0 7.0; 9.0 11.0; 13.0 15.0])"
+constraint = "[no_constraint(), no_constraint(), bounded_below(-2.0), bounded_above(100.0)]"
+type = "float"
+
+# Convenience notation for arrays of parameters that all have the same prior 
+# distribution and constraints. 
+# `repeat([prior], n_dims)` will construct `n_dims` prior distribution. 
+# Similarly,  `repeat([constraint], n_dims)` will construct `n_dims` 
+# constraints. 
+# The resulting `n_dims` parameters will automatically be attaching numbers, 
+# e.g., "uq_param_6_(1)", "uq_param_6_(2)", etc.
+[uq_param_6]
+prior = "repeat([Parameterized(Gamma(2.0, 1.0))], 3)"
+constraint = "repeat([bounded_above(9.0)], 3)"
+type = "float"
+
+# A repeated constraint can also be combined with a multivariate distribution.
+[uq_param_7]
+prior = "Parameterized(MvNormal(3, 2.0))"
+constraint = "repeat([no_constraint()], 3)"
```

Comment:
I presumed that 
`constraint = "repeat([no_constraint()], 3)"` would interpret as
`[[no_constraint()], [no_constraint()], [no_constraint()] ]` 
(which is what appears to happen with `UQ_param_6`)

And `constraint = "repeat(no_constraint(), 3)"` would be 
`[no_constraint(), no_constraint(), no_constraint()] ` 
But this appears to give an error for `UQ_param_7`?

Reply:
Yes, this was somewhat inconsistent. I changed the specification of repeated constraints in the toml files to

```
[uq_param_6]
prior = "repeat([Parameterized(Gamma(2.0, 1.0))], 3)"
constraint = "repeat([[bounded_above(9.0)]], 3)"

[uq_param_7]
prior = "Parameterized(MvNormal(3, 2.0))"
constraint = "repeat([no_constraint()], 3)"
``` 

The idea here is that the "repeat(...)" expression for the constraint should be valid Julia code (without the quotation marks of course), such that the corresponding `ParameterDistributions` could be constructed e.g. in a REPL according to:

```
# parameter distribution for uq_param_6 (note that the parameters are generated internally here)
pd6 = ParameterDistribution(
        repeat([Parameterized(Gamma(2.0, 1.0))], 3), 
        repeat([[bounded_above(9.0)]], 3), 
        ["uq_param_6_(1)", "uq_param_6_(2)", "uq_param_6_(3)"])

# parameter distribution for uq_param_7
pd7 = ParameterDistribution(
        Parameterized(MvNormal(3, 2.0)),
        repeat([no_constraint()], 3),
        "uq_param_7")
```
I first wanted to simplify the writing of these expressions in the toml (e.g., by not having the user write too many square brackets :-)), but have come to think that it is much simpler to require the expressions to be written the same way you would write them if you were to actually call the `ParameterDistribution` constructor. This also allows the user to test the validity of the expressions by simply trying to pass them to the `ParameterDistribution` constructor and checking whether that throws an error or not.